Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/derivetime/TimeIntervalDeriver.kt

```diff
@@ -0,0 +1,77 @@
+package de.rki.coronawarnapp.eventregistration.checkins.derivetime
+
+import de.rki.coronawarnapp.appconfig.PresenceTracingSubmissionParamContainer
+import de.rki.coronawarnapp.risk.DefaultRiskLevels.Companion.inRange
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+import kotlin.math.max
+import kotlin.math.roundToLong
+
+private val INTERVAL_LENGTH_IN_SECONDS = TimeUnit.MINUTES.toSeconds(10L)
+
+private fun alignToInterval(timestamp: Long) =
+    (timestamp / INTERVAL_LENGTH_IN_SECONDS) * INTERVAL_LENGTH_IN_SECONDS
+
+/**
+ * Derive CheckIn start and end times
+ * @param startTimestampInSeconds [Long] timestamp in seconds
+ * @param endTimestampInSeconds [Long] timestamp in seconds
+ */
+fun PresenceTracingSubmissionParamContainer.deriveTime(
+    startTimestampInSeconds: Long,
+    endTimestampInSeconds: Long
+): Pair<Long, Long>? {
+    val durationInSeconds = max(0, endTimestampInSeconds - startTimestampInSeconds)
+    Timber.d("durationInSeconds: $durationInSeconds")
+
+    val durationInMinutes = TimeUnit.SECONDS.toMinutes(durationInSeconds)
+    Timber.d("durationInMinutes: $durationInMinutes")
+
+    val dropDueToDuration = durationFilters.any { durationFilter ->
```

Comment:
Shouldn't we specific the specific type to be `Long` to not be dependent on the type of `durationFilter`?
Same for `aerosole ` below

Reply:
Can you please elaborate more on that? do you mean to specify the type for `dropDueToDuration` ?
such as `val dropDueToDuration:Boolean`

Reply:
done